grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcServiceImplErrorTest.java

```diff
@@ -0,0 +1,432 @@
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            final GrpcService grpcService = GrpcService.builder()
+                                                       .addService(new ServerImplErrorAtMetadataService())
+                                                       .build();
+//            sb.service('/' + TestServiceGrpc.getHalfDuplexCallMethod().getFullMethodName(), grpcService);
+//            sb.service('/' + TestServiceGrpc.getFullDuplexCallMethod().getFullMethodName(), grpcService);
+            sb.service(grpcService);
+//            sb.maxConnectionAgeMillis(3000); // cut off connection for the case when server side error.
+            sb.decorator(LoggingService.newDecorator());
+        }
+    };
+
+    @RegisterExtension
+    static ServerExtension serverWithBlockingExecutor = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            final GrpcService grpcService = GrpcService.builder()
+                                                       .addService(new ServerImplErrorAtMetadataService())
+                                                       .enableUnframedRequests(true)
+                                                       .supportedSerializationFormats(
+                                                               GrpcSerializationFormats.values())
+                                                       .useBlockingTaskExecutor(true)
+                                                       .build();
+//            sb.service('/' + TestServiceGrpc.getUnaryCallMethod().getFullMethodName(), grpcService);
+//            sb.service('/' + TestServiceGrpc.getStreamingOutputCallMethod().getFullMethodName(), grpcService);
+            sb.service(grpcService);
+//            sb.maxConnectionAgeMillis(3000); // cut off connection for the case when server side error.
+            sb.decorator(LoggingService.newDecorator());
+        }
+    };
+    private static ManagedChannel channel;
+    private static ManagedChannel blockingChannel;
+
+    @BeforeAll
+    static void setUpChannel() {
+        channel = ManagedChannelBuilder.forAddress("127.0.0.1", server.httpPort())
+                                       .usePlaintext()
+                                       .build();
+        blockingChannel = ManagedChannelBuilder.forAddress("127.0.0.1", serverWithBlockingExecutor.httpPort())
+                                               .usePlaintext()
+                                               .build();
+    }
+
+    @AfterAll
+    static void tearDownChannel() {
+        channel.shutdownNow();
+        blockingChannel.shutdownNow();
+    }
+
+    @Test
+    void clientUnaryCall2() throws InterruptedException {
+        Throwable throwable = catchThrowable(() -> {
+            TestServiceBlockingStub service = TestServiceGrpc.newBlockingStub(blockingChannel);
```

Comment:
We're requesting to `blockingChannel` (which points to `serverWithBlockingExecutor`), but we're checking for the request context for `server` at L136.

I think we want to unify check the request context for the correct server

```suggestion
            TestServiceBlockingStub service = TestServiceGrpc.newBlockingStub(channel);
```

Reply:
I think once this change is made unary server call tests are passing without a problem. Can you also check? 😄

Reply:
Yes, that's right.
My brain was fog when I have implemented.
It's resolved now 🙇